modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/ast/ApplicationExtensionModelsEnricher.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.ast;
+
+import static org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType;
+import static org.mule.runtime.api.component.TypedComponentIdentifier.ComponentType.OPERATION_DEF;
+import static org.mule.runtime.api.dsl.DslResolvingContext.getDefault;
+import static org.mule.runtime.core.api.util.boot.ExtensionLoaderUtils.getOptionalLoaderById;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_ARTIFACT_AST_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_EXTENSION_NAME_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.ExtensionConstants.MULE_SDK_APPLICATION_LOADER_ID;
+import static org.mule.runtime.extension.api.ExtensionConstants.VERSION_PROPERTY_NAME;
+import static org.mule.runtime.extension.api.loader.ExtensionModelLoadingRequest.builder;
+
+import static java.util.Collections.singleton;
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.ArtifactType;
+import org.mule.runtime.extension.api.loader.ExtensionModelLoader;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+
+/**
+ * An {@link ArtifactExtensionModelsEnricher} suitable for the context of applications.
+ *
+ * @since 4.5.0
+ */
+public class ApplicationExtensionModelsEnricher implements ArtifactExtensionModelsEnricher {
+
+  private static final Logger LOGGER = getLogger(ArtifactAstUtils.class);
+
+  // TODO W-11796759: This class shouldn't know which are the specific reusable components.
+  private static final Set<ComponentType> REUSABLE_COMPONENT_TYPES = singleton(OPERATION_DEF);
+
+  private final String artifactId;
+  private final Optional<String> artifactVersion;
+
+  public ApplicationExtensionModelsEnricher(String artifactId, Optional<String> artifactVersion) {
+    this.artifactId = artifactId;
+    this.artifactVersion = artifactVersion;
+  }
+
+  @Override
+  public boolean applicable(ArtifactAst ast) {
```

Comment:
This method is part of the `ArtifactExtensionModelsEnricher` (I'm in the process of choosing a better name for it).
Even though the user of this specific concrete class will know that it is used in the context of an application, [the code that consumes the instance](https://github.com/mulesoft/mule/pull/11776/files#diff-8a588bb1806dd53f441fcbda4dfe927efa22ad5e5deeabe5126e0016b41ab4e2R52) will not know if it is happening in the context of an application or an extension. It only sees the interface and not the concrete class.
Still, I'm considering refactoring this class so that it returns an `Optional<ExtensionModel>` instead of an "enriched" `Set<ExtensionModel>`. If I do that, I think I could make this a private method and call it directly from the parsing/loading method.
For now I renamed it to `isApplicable` as suggested.

Reply:
Removed.